scapy/contrib/pnio.py

```diff
@@ -14,67 +15,374 @@
 
 # Copyright (C) 2016 Gauthier Sebaux
 
-# scapy.contrib.description = ProfinetIO base layer
+# scapy.contrib.description = ProfinetIO RTC (+Profisafe) layer
 # scapy.contrib.status = loads
+import copy
+from scapy.compat import raw
+from scapy.error import Scapy_Exception
+from scapy.config import conf
+from scapy.packet import Packet, bind_layers
+from scapy.layers.l2 import Ether
+from scapy.layers.inet import UDP
+from scapy.fields import (
+    XShortEnumField, BitEnumField, XBitField,
+    BitField, StrField, PacketListField,
+    StrFixedLenField, ShortField,
+    FlagsField, ByteField, XIntField, X3BytesField
+)
 
-"""
-A simple and non exhaustive Profinet IO layer for scapy
-"""
 
-# Scapy imports
-from __future__ import absolute_import
-from scapy.all import Packet, bind_layers, Ether, UDP
-from scapy.fields import XShortEnumField
-from scapy.modules.six.moves import range
-
-# Some constants
 PNIO_FRAME_IDS = {
-    0x0020: "PTCP-RTSyncPDU-followup",
-    0x0080: "PTCP-RTSyncPDU",
-    0xFC01: "Alarm High",
-    0xFE01: "Alarm Low",
-    0xFEFC: "DCP-Hello-Req",
-    0xFEFD: "DCP-Get-Set",
-    0xFEFE: "DCP-Identify-ReqPDU",
-    0xFEFF: "DCP-Identify-ResPDU",
-    0xFF00: "PTCP-AnnouncePDU",
-    0xFF20: "PTCP-FollowUpPDU",
-    0xFF40: "PTCP-DelayReqPDU",
-    0xFF41: "PTCP-DelayResPDU-followup",
-    0xFF42: "PTCP-DelayFuResPDU",
-    0xFF43: "PTCP-DelayResPDU",
+    0x0020: 'PTCP-RTSyncPDU-followup',
+    0x0080: 'PTCP-RTSyncPDU',
+    0xFC01: 'Alarm High',
+    0xFE01: 'Alarm Low',
+    0xFEFC: 'DCP-Hello-Req',
+    0xFEFD: 'DCP-Get-Set',
+    0xFEFE: 'DCP-Identify-ReqPDU',
+    0xFEFF: 'DCP-Identify-ResPDU',
+    0xFF00: 'PTCP-AnnouncePDU',
+    0xFF20: 'PTCP-FollowUpPDU',
+    0xFF40: 'PTCP-DelayReqPDU',
+    0xFF41: 'PTCP-DelayResPDU-followup',
+    0xFF42: 'PTCP-DelayFuResPDU',
+    0xFF43: 'PTCP-DelayResPDU',
```

Comment:
Those changes are a bit un-necessary :/
Also, I prefer double quotes (but that's not the point)

Reply:
Is there a coding convention for Scapy regarding ' and " ?
I just usually put single quotes everywhere as it make the code "lighter" to read (less quotes are displayed for the same result).

Anyway I'll undo it at least for the dict déclaration

Reply:
Thanks !

There is no convention about that :/ Scapy is full of both